src/shared/lib/resources/types.ts

```diff
@@ -558,6 +558,15 @@ export const ScriptEventPreset = Type.Object({
 
 export type ScriptEventPreset = Static<typeof ScriptEventPreset>;
 
+export const CollisionSetting = Type.Object({
+  key: Type.String(),
+  name: Type.Optional(Type.String()),
+  color: Type.String(),
+  icon: Type.Optional(Type.String()),
```

Comment:
I never ended up using this, but the plan was show a custom image. Instead, I just use the first character of name for the icon.